modules/tts/tts-adapters/tts-adapter-cereproc/src/main/java/org/daisy/pipeline/tts/cereproc/impl/util/UCharReplacer.java

```diff
@@ -0,0 +1,195 @@
+package org.daisy.pipeline.tts.cereproc.impl.util;
+/*
+ * org.daisy.util (C) 2005-2008 Daisy Consortium
+ * 
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * 
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import java.net.URL;
+import java.text.Normalizer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Substitute unicode characters with replacement strings.
+ *
+ * <p>The substitution is made using different attempts in a series of preference;
+ * each successor is considered a fallback to its predecessor.</p>
+ * <ol>
+ *   <li>Locate a replacement string in one or several user provided tables;</li>
+ * </ol>
+ * 
+ * <p>The use of this class <em>may</em> result in a change in unicode character composition between input and output.
+ * If you need a certain normalization form, normalize after the use of this class.</p>
+ * 
+ * <p>Usage example:</p>
+ * <code><pre>
+ * 	UCharReplacer ucr = new UCharReplacer();
+ *	ucr.addTranslationTable(fileURL);
+ *  ucr.addTranslationTable(fileURL2);
+ *	String ret = ucr.toReplacementString(input);
+ * </pre></code>
+ *  
+ * <p>The translation table file is using the same xml format as that of
+ * java.util.Properties [1][2], using the HEX representation (without 
+ * the characteristic 0x-prefix!) of a unicode character as the <tt>key</tt> 
+ * attribute and the replacement string as value of the <tt>entry</tt> 
+ * element.</p>
+ * 
+ * <p>If the <tt>key</tt> attribute contains exactly one unicode codepoint 
+ * (one character) it will be treated literally. It will not be interpreted 
+ * as a HEX representation of another character, even if theoretically 
+ * possible. E.g. if the <tt>key</tt> is "a", it will be
+ * treated as 0x0061 rather than as 0x000a</p>
+ * 
+ * <p>Note - there is a significant difference between a unicode codepoint (32 bit int)
+ * and a UTF16 codeunit (=char) - a codepoint consists of one or two codeunits.</p>  
+ * <p>To make sure an int represents a codepoint and not a codeunit.
```

Comment:
This sentence got cut short?

Reply:
oh sorry, thats a mistake. I was gonna remove parts mentioning com.ibm.icu.text-related stuff. I will fix that